Feature: Integrate WebPack in Bundle

ceesvanegmond opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->

## Title
Integrate WebPack in this bundle. Remove all compiled stuff and build them your self.

## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Added
- Added webpack config file

### Removed
- gulpfile.js and gulp dependencies
```

## To do

<!--
    If this is a work in progress, COMPLETE and ADD needed tasks.
    You can add as many tasks as you want.
    If some are not relevant, just REMOVE them.
-->

- [x] Update the documentation
- [x] Add an upgrade note

## Subject

<!-- Describe your Pull Request content here -->

ceesvanegmond committed:
Feature: First try to integrate webpack in bundle

Emroni commented on webpack.config.js:
```diff
@@ -0,0 +1,73 @@
+        new Webpack.optimize.UglifyJsPlugin({
+            compress: true,
+            sourceMap: false
+        }),
```
We've noticed that using the minimize and compress here results in differences in results sometimes. Rather remove lines 60-67.

ceesvanegmond replied:
Also from devDependencies? I quess ```        "@wearejust/trace": "^2.0.5",
``` must be in there right?

ceesvanegmond replied:
@Emroni No minimize at all?

Emroni commented on package.json:
```diff
@@ -1,22 +1,19 @@
+        "trace.js": "^2.0.0",
+        "webpack": "^2.4.1"
+    }
 }
```
No devDependencies?

Emroni replied:
One other thing: both wearejust/trace and trace.js can be removed from the dependencies.

ceesvanegmond replied:
Also from devDependencies? I quess ```"@wearejust/trace": "^2.0.5",``` must be in there right?

Emroni replied:
Sure, can be. Looks good to me :)

ceesvanegmond committed:
Feature: Better WebPack config

ceesvanegmond committed:
Merge branch '1.x' into feature/webpack

ceesvanegmond committed:
Feature: Add build folder but ignore all in it

ceesvanegmond committed:
Chore: Update README

ceesvanegmond committed:
Feature: Add new path to build

ceesvanegmond committed:
Feature: Add versioning with webpack and manifest.json

ceesvanegmond committed:
Chore: Add stuff to README on how to use in combinatio with manifest.json

ceesvanegmond committed:
Feature: Update package.json to standards

ceesvanegmond committed:
Feature: Add production mode

ceesvanegmond committed:
Merge branch '1.x' into feature/webpack

ceesvanegmond committed:
Merge branch '1.x' into feature/webpack

ceesvanegmond commented:
@Emroni @johnhout I will target new 2.x branch since this is an BC.

ceesvanegmond committed:
Feature: Add Upgrade guide

ceesvanegmond committed:
Chore: Add filenames to upgrade guide

ceesvanegmond committed:
Update UPGRADE-2.0.md

ceesvanegmond committed:
Update UPGRADE-2.0.md

ceesvanegmond committed:
Update UPGRADE-2.0.md

ceesvanegmond committed:
Update README.md

ceesvanegmond committed:
Update composer.json

Add Symfony Framework 3.3 as minimum requirement because of the [json_manifest_path](https://symfony.com/blog/new-in-symfony-3-3-manifest-based-asset-versioning)

ceesvanegmond committed:
Update UPGRADE-2.0.md

ceesvanegmond committed:
Update UPGRADE-2.0.md
